Sources/NIOPosix/SelectorGeneric.swift

```diff
@@ -43,10 +43,18 @@ extension Optional {
 extension timespec {
     @inlinable
     init(timeAmount amount: TimeAmount) {
+        #if canImport(wasi_pthread)
+        typealias _Seconds = Int64
+        #elseif canImport(Glibc)
+        typealias _Seconds = time_t
+        #else
+        typealias _Seconds = Int
+        #endif
+
         let nsecPerSec: Int64 = 1_000_000_000
         let ns = amount.nanoseconds
         let sec = ns / nsecPerSec
-        self = timespec(tv_sec: Int(sec), tv_nsec: Int(ns - sec * nsecPerSec))
+        self = timespec(tv_sec: _Seconds(sec), tv_nsec: Int(ns - sec * nsecPerSec))
```

Comment:
Similarly, I'm inclined to assume we could just make this a `time_t` on all platforms.